pkg/share/options.go

```diff
@@ -0,0 +1,25 @@
+package share
+
+import (
+	"context"
+	"github.com/justtrackio/gosoline/pkg/cfg"
+	"github.com/justtrackio/gosoline/pkg/kernel"
+	"github.com/justtrackio/gosoline/pkg/log"
+)
+
+func KernelMiddlewareShares(ctx context.Context, config cfg.Config, logger log.Logger, next kernel.Handler) kernel.Handler {
+	return func() {
+		manager, err := ProvideShareManager(ctx, config, logger)
+		if err != nil {
+			logger.Error("can not create share manager: %s", err.Error())
+			return
+		}
+
+		if err := manager.SetupShareTable(); err != nil {
+			logger.Error(err.Error())
+			return
+		}
+
+		next()
+	}
+}
```

Comment:
I'm not up-to-date RN with gosoline and middlewares. Is returning in case of an error the right thing to do or should the next middleware be executed?